scripts/stac/imagery/item.py

```diff
@@ -26,17 +28,59 @@
 class ImageryItem:
     stac: dict[str, Any]
 
-    def __init__(self, id_: str, now_string: str, stac_asset: STACAsset, stac_processing: STACProcessing) -> None:
+    def __init__(self, id_: str, stac_asset: STACAsset, stac_processing: STACProcessing) -> None:
         self.stac = {
             "type": "Feature",
             "stac_version": STAC_VERSION,
             "id": id_,
             "links": [Link(path=f"./{id_}.json", rel=Relation.SELF, media_type=StacMediaType.GEOJSON).stac],
             "assets": {"visual": {**stac_asset, "type": "image/tiff; application=geotiff; profile=cloud-optimized"}},
             "stac_extensions": [StacExtensions.file.value, StacExtensions.processing.value],
-            "properties": {"created": now_string, "updated": now_string, **stac_processing},
+            "properties": {"created": stac_asset["created"], "updated": stac_asset["updated"], **stac_processing},
         }
 
+    @classmethod
+    def from_file(cls, file_name: str) -> "ImageryItem":
+        """Create an ImageryItem from a file.
+
+        Args:
+            file_name: The s3 URL or local path of the file to load.
+
+        Returns:
+            ImageryItem: The new ImageryItem.
+        """
+        file_content = read(file_name)
+        stac_dict_from_s3 = json.loads(file_content.decode("UTF-8"))
+        if "bbox" in stac_dict_from_s3:  # tuple() does not serialise well into json
+            stac_dict_from_s3["bbox"] = tuple(stac_dict_from_s3["bbox"])
+        new_item = cls(
+            id_=stac_dict_from_s3["id"],
+            stac_asset=stac_dict_from_s3["assets"]["visual"],
+            stac_processing=stac_dict_from_s3["properties"],
+        )
+        new_item.stac = stac_dict_from_s3
+
+        return new_item
+
+    def set_checksum(self, file_content_checksum: str, stac_processing_data: STACProcessing) -> None:
```

Comment:
Rather than passing a `STACProcessing` object, I'd probably just pass the attributes of the object which are needed for this method. That way this method doesn't need to know about the structure of `STACProcessing`, and we refactor it easier.

Reply:
You could do something like `for key, value in stac_processing_data.items() if key != "processing:datetime":`, so that this method doesn't need to know anything else about the structure of the object?